shell/platform/linux/fl_basic_message_channel_test.cc

```diff
@@ -222,3 +222,35 @@ TEST(FlBasicMessageChannelTest, SendNullMessageWithResponse) {
   // Blocks here until null_message_response_cb is called.
   g_main_loop_run(loop);
 }
+
+// Called when the message response is received in the CustomType test.
+static void custom_type_response_cb(GObject* object,
+                                    GAsyncResult* result,
+                                    gpointer user_data) {
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(FlValue) message = fl_basic_message_channel_send_finish(
+      FL_BASIC_MESSAGE_CHANNEL(object), result, &error);
+  EXPECT_EQ(message, nullptr);
+  EXPECT_NE(error, nullptr);
+  EXPECT_STREQ(error->message, "Custom value not implemented");
+
+  g_main_loop_quit(static_cast<GMainLoop*>(user_data));
+}
+
+// Checks sending a message with a custom type generates an error.
+TEST(FlBasicMessageChannelTest, CustomType) {
+  g_autoptr(GMainLoop) loop = g_main_loop_new(nullptr, 0);
+
+  // Attempt to send an integer with the string codec.
+  g_autoptr(FlEngine) engine = make_mock_engine();
+  g_autoptr(FlBinaryMessenger) messenger = fl_binary_messenger_new(engine);
+  g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+  g_autoptr(FlBasicMessageChannel) channel = fl_basic_message_channel_new(
+      messenger, "test/echo", FL_MESSAGE_CODEC(codec));
+  g_autoptr(FlValue) message = fl_value_new_custom(42, nullptr, nullptr);
+  fl_basic_message_channel_send(channel, message, nullptr,
+                                custom_type_response_cb, loop);
+
+  // Blocks here until custom_type_response_cb is called.
+  g_main_loop_run(loop);
```

Comment:
May be worth adding a timeout to ensure it doesn't hang indefinitely if the callback isn't called for any reason.

Reply:
This is a common pattern in the tests. If it did block forever, the test framework would cancel the test. I think given complexity in adding the timeout it would not be worth it (it's only useful in a failed test case).